pastas/stats/signatures.py

```diff
@@ -0,0 +1,1353 @@
+    """Baseflow index according to [wmo_2008]_.
+
+    Parameters
+    ----------
+    series: pandas.Series
+        Pandas Series with DatetimeIndex and head values.
+    normalize: bool, optional
+        normalize the time series to values between zero and one.
+
+    Returns
+    -------
+    float
+
+    Notes
+    -----
+    Adapted analogously to its application in streamflow. Here, a baseflow
+    time series is separated from a 5‐day minimum groundwater head in a
+    moving window. BFI equals the total sum of heads of original time series
+    divided by the total sum of heads from the baseflow type of time series.
+
+    References
+    ----------
+    .. [wmo_2008] WMO (2008). Manual on Low‐Flow Estimation and Prediction.
+       Geneva, Switzerland: World Meteorological Organization.
+
+    """
+    if normalize:
+        series = _normalize(series)
+
+    # A/B. Selecting minima hm over 5 day periods
+    hm = series.resample("5D").min()
+
+    # C. define the turning points ht (0.9 * head < adjacent heads)
+    ht = pd.Series(dtype=float)
+    for i, h in enumerate(hm.iloc[1:-1], start=1):
+        if (h < hm.iloc[i - 1]) & (h < hm.iloc[i + 1]):
+            ht[hm.index[i]] = h
+
+    # D. Interpolate
+    ht = ht.resample("D").interpolate()
+
+    # E. Assign a base head to each day
+    ht[ht > series.resample("D").mean().loc[ht.index]] = \
+        series.resample("D").mean()
+
+    # Compute the baseflow index
+    bfi = series.resample("D").mean().sum() / ht.sum()
+    return bfi
+
+
+def baseflow_stability(series, normalize=True):
+    """Baseflow stability after [heudorfer_2019]_.
+
+    Parameters
+    ----------
+    series: pandas.Series
+        Pandas Series with DatetimeIndex and head values.
+    normalize: bool, optional
+        normalize the time series to values between zero and one.
+
+    Returns
+    -------
+    float
+
+    Notes
+    -----
+    Originally developed for streamflow, here the Base Flow Index
+    algorithm is analogously adapted to groundwater time series as a filter
+    to separate the slow component (“baseflow”) of the time series. Then,
+    the mean annual baseflow is calculated. Base Flow Stability is the
+    difference of maximum and minimum annual baseflow.
+
+    References
+    ----------
+    .. [heudorfer_2019] Heudorfer, B., Haaf, E., Stahl, K., & Barthel, R.
+       (2019). Index‐based characterization and quantification of groundwater
+       dynamics. Water Resources Research, 55, 5575–5592.
+
+    """
+    if normalize:
+        series = _normalize(series)
+
+    # A/B. Selecting minima hm over 5 day periods
+    hm = series.resample("5D").min()
+
+    # C. define the turning points ht (0.9 * head < adjacent heads)
+    ht = pd.Series(dtype=float)
+    for i, h in enumerate(hm.iloc[1:-1], start=1):
+        if (h < hm.iloc[i - 1]) & (h < hm.iloc[i + 1]):
+            ht[hm.index[i]] = h
+
+    # D. Interpolate
+    ht = ht.resample("D").interpolate()
```

Comment:
[TypeError: Only valid with DatetimeIndex, TimedeltaIndex or PeriodIndex, but got an instance of 'Index']()

Seems like `ht` is missing a datetime index so the resampling does not work.

Reply:
Got this error while testing:
```
rd = pd.date_range('01-01-2020', '31-12-2020', freq='D')
series = pd.Series(np.linspace(0, 100, len(rd)), index=rd)
baseflow_index(series, normalize=False)
```

Reply:
That is because you made a mistake in creating the Index. rd is not a DatetimeIndex because you don't provide proper string formats ;-)